src/rtc.c

```diff
@@ -945,8 +946,15 @@ void RTC_StartAlarm(alarm_t name, uint8_t day, uint8_t hours, uint8_t minutes, u
 #if defined(RTC_ICSR_BIN)
     if ((initMode == MODE_BINARY_ONLY) || (initMode == MODE_BINARY_MIX)) {
       /* We have an SubSecond alarm to set in RTC_BINARY_MIX or RTC_BINARY_ONLY mode */
-      /* The subsecond in ms is converted in ticks unit 1 tick is 1000 / fqce_apre */
-      RTC_AlarmStructure.AlarmTime.SubSeconds = UINT32_MAX - (subSeconds * (predivSync + 1)) / 1000;
+      /* The subsecond in ms is converted in ticks unit 1 tick is 1000 / fqce_apre
+       * It keeps the subsecond accuracy on 64 bits if needed
+       */
+      if (subSeconds > (uint64_t)UINT32_MAX) {
+        uint64_t tmp = ((uint64_t)subSeconds * (predivSync + 1)) / (uint64_t)1000;
```

Comment:
Is the uint64_t cast of subseconds needed  as it is already a uint64_t?

Reply:
> Is the uint64_t cast of subseconds needed as it is already a uint64_t?
maybe 
`uint64_t tmp = (subSeconds * (uint64_t)(predivSync + 1)) / (uint64_t)1000;`

Reply:
changed in that way